soroban-sdk/src/env.rs

```diff
@@ -509,6 +510,75 @@ impl Env {
             .unwrap()
     }
 
+    /// Register the built-in Stellar Asset Contract with provided admin address.
+    ///
+    /// Returns the contract ID of the registered token contract.
+    ///
+    /// The contract will wrap a randomly-generated Stellar asset. This function
+    /// is useful for using in the tests when an arbitrary token contract
+    /// instance is needed.
+    pub fn register_stellar_asset_contract_with_admin(&self, admin: Address) -> BytesN<32> {
+        let issuer_id =
+            xdr::AccountId(xdr::PublicKey::PublicKeyTypeEd25519(xdr::Uint256(random())));
+
+        self.host()
+            .with_mut_storage(|storage| {
+                let k = xdr::LedgerKey::Account(xdr::LedgerKeyAccount {
+                    account_id: issuer_id.clone(),
+                });
+
+                if !storage.has(
+                    &k,
+                    soroban_env_host::budget::AsBudget::as_budget(self.host()),
+                )? {
+                    let v = xdr::LedgerEntry {
+                        data: xdr::LedgerEntryData::Account(xdr::AccountEntry {
+                            account_id: issuer_id.clone(),
+                            balance: 0,
+                            flags: 0,
+                            home_domain: xdr::StringM::default(),
+                            inflation_dest: None,
+                            num_sub_entries: 0,
+                            seq_num: xdr::SequenceNumber(0),
+                            thresholds: xdr::Thresholds([1; 4]),
+                            signers: xdr::VecM::default(),
+                            ext: xdr::AccountEntryExt::V0,
+                        }),
+                        last_modified_ledger_seq: 0,
+                        ext: xdr::LedgerEntryExt::V0,
+                    };
+                    storage.put(
+                        &k,
+                        &v,
+                        soroban_env_host::budget::AsBudget::as_budget(self.host()),
+                    )?
+                }
+                Ok(())
+            })
+            .unwrap();
```

Comment:
If we kept the Accounts testutils we could use them to create this account rather than embedding it into the registration.

Reply:
Agree, but I'm not sure if we should keep these as is, as there is really no other use for them yet. Let's clean this up when we figured out when/if the accounts may be needed.